### PR TITLE
Engines

### PR DESCRIPTION
**Pull request by robertzk**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ install:
 - "./travis-tool.sh install_github robertzk/testthatsomemore"
 - "./travis-tool.sh install_github robertzk/objectdiff"
 - "./travis-tool.sh install_github robertzk/stagerunner"
-- "./travis-tool.sh install_github robertzk/director"
+- "./travis-tool.sh install_github robertzk/director@documentation_cleanup"
```
Comment (peterhurford) [inline .travis.yml]: When can we get this part merged? ;)

File: .travis.yml [hunk 2]
```diff
@@ -35,5 +35,9 @@ notifications:
     template:
     - "%{repository}#%{build_number} (%{branch} - %{commit} : %{author}): %{message}
       | Details: %{build_url} | Changes: %{compare_url}"
-    rooms:
-      secure: l+wmlZJjYTD9Gv+oSp+y5IE6UFAwGEbfiUdoIU8VA4AihIsarEu9/h1WfuZLf00qlxktx45BLsecvn5QGX+IHVTXDFxmYyL+pzuTdNYZjc87EutstyJ1Ahdt5IA49A8X+PHAO7v60XIjrUrXPKwx3vxsVmO1RSayo0xAwfnupmk=
+  rooms:
+    secure: l+wmlZJjYTD9Gv+oSp+y5IE6UFAwGEbfiUdoIU8VA4AihIsarEu9/h1WfuZLf00qlxktx45BLsecvn5QGX+IHVTXDFxmYyL+pzuTdNYZjc87EutstyJ1Ahdt5IA49A8X+PHAO7v60XIjrUrXPKwx3vxsVmO1RSayo0xAwfnupmk= 
+env:
+  - global:
+    - WARNINGS_ARE_ERRORS=1
```
Comment (peterhurford) [inline .travis.yml]: Hard mode.

File: DESCRIPTION [hunk 3]
```diff
@@ -14,27 +14,24 @@ Authors@R: c(person("Robert", "Krzyzanowski",
...
     statsUtils,
     mungebits,
-    Matrix,
     tundra,
     syberiaMungebits,
     syberiaStructure,
     syberiaStages,
-    objectdiff
+    objectdiff,
+    R6 (>= 2.1.0)
```
Comment (peterhurford) [inline DESCRIPTION]: How are you ordering your dependencies?

File: .travis.yml (hunk 1 above)
Comment (robertzk) [inline .travis.yml]: There will be a big push :)

File: .travis.yml (hunk 2 above)
Comment (robertzk) [inline .travis.yml]: :+1:

File: R/director.R [hunk 4]
```diff
@@ -311,15 +325,14 @@ default_tests_preprocessor <- function(resource_object, director, source_args, s
   tested_resource <- gsub("^test\\/", "", resource)
   if (!director$exists(tested_resource)) {
     # TODO: (RK) Figure out how this interacts with virtual resources.
-    #warning("You are testing ", sQuote(director:::colourise(tested_resource, "yellow")),
+    #warning("You are testing ", sQuote(crayon::yellow(tested_resource)),
     #        " but it does not exist in the project.\n", call. = FALSE, immediate = TRUE)
     #return(NULL)
```
Comment (robertzk) [inline R/director.R]: Not ready yet :(

**Pull request closed without merging**